recirq/quantum_chess/quantum_board.py

```diff
@@ -255,13 +258,19 @@ def sample_with_ancilla(self, num_samples: int
                         f'Discarded {error_count} from error mitigation '
                         f'{noise_count} from noise and '
                         f'{post_count} from post-selection\n')
+                    t1 = time.perf_counter()
+                    self.debug_log += (f"sample_with_ancilla takes {t1 - t0:0.4f} seconds.\n")
+                    self.timing_stats.append(t1 - t0)
```

Comment:
Should we separate this into a function, since it is called twice?  That will make it easier to keep both branches in sync.